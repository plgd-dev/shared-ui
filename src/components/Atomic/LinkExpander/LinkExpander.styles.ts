import { css } from '@emotion/react'
import { ThemeType, getThemeColor } from '../_theme'

export const expandContent = (theme: ThemeType) => css`
    padding: 24px;
    border-radius: 8px;
    background: ${getThemeColor(theme, `LinkExpander.background`)};
`
