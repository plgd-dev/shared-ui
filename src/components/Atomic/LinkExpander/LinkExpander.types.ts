import { ReactNode } from 'react'

export type Props = {
    children: ReactNode
    i18n: {
        hide: string
        show: string
        name: string
    }
    show: boolean
    toggleView: (view: boolean) => void
}
