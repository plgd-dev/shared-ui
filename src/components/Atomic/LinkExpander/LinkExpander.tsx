import React, { FC } from 'react'
import { AnimatePresence, motion } from 'framer-motion'

import Link from '../Link'
import * as styles from './LinkExpander.styles'

const LinkExpander: FC<any> = (props) => {
    const { children, i18n, show, toggleView } = props

    const AnimatedLink = () => (
        <AnimatePresence>
            <motion.div
                animate={{ opacity: 1 }}
                exit={{
                    opacity: 0,
                }}
                initial={{ opacity: 0 }}
                transition={{
                    duration: 0.3,
                }}
            >
                <Link
                    href='#'
                    onClick={(e) => {
                        e.preventDefault()
                        toggleView(!show)
                    }}
                    size='big'
                >
                    {`${show ? i18n.hide : i18n.show} ${i18n.name}`}
                </Link>
            </motion.div>
        </AnimatePresence>
    )

    return (
        <div>
            <AnimatePresence>
                {show && (
                    <motion.div
                        animate={{ opacity: 1, height: 'auto' }}
                        css={styles.expandContent}
                        exit={{
                            opacity: 0,
                            height: 0,
                        }}
                        initial={{ opacity: 0, height: 0 }}
                        transition={{
                            duration: 0.3,
                        }}
                    >
                        <>
                            {children}
                            {show && <AnimatedLink />}
                        </>
                    </motion.div>
                )}
            </AnimatePresence>
            {!show && <AnimatedLink />}
        </div>
    )
}

LinkExpander.displayName = 'LinkExpander'

export default LinkExpander
