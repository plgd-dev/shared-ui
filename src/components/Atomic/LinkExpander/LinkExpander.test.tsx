import { render } from '@testing-library/react'
import LinkExpander from './LinkExpander'

describe('<LinkExpander>', () => {
    it('renders without crashing', () => {
        const { container, asFragment } = render(
            <LinkExpander i18n={{ show: 'Show', hide: 'Hide', name: 'Name' }} show={false} toggleView={() => {}}>
                Test
            </LinkExpander>
        )
        expect(container).toBeInTheDocument()
        expect(asFragment()).toMatchSnapshot()
    })

    it('displays show text when not expanded', () => {
        const { getByText } = render(
            <LinkExpander i18n={{ show: 'Show', hide: 'Hide', name: 'Name' }} show={false} toggleView={() => {}}>
                Test
            </LinkExpander>
        )
        expect(getByText('Show Name')).toBeInTheDocument()
    })
})
