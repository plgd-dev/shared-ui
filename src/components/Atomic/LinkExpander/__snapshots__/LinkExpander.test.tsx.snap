// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkExpander> renders without crashing 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-bottom: 2px solid;
  -webkit-text-decoration: none!important;
  text-decoration: none!important;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  padding-bottom: 2px;
  font-weight: bold;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom-color: transparent;
}

<div>
    <div
      style="opacity: 0;"
    >
      <a
        class="emotion-0"
        href="#"
      >
        Show Name
      </a>
    </div>
  </div>
</DocumentFragment>
`;
